### PR TITLE
	Support expressions in rest arg access

### PR DESCRIPTION
**Pull request by amasad**
The current implementation assumes a numeric literal although it just
checks the base type.

### CODE REVIEW

File: packages/babel-plugin-transform-es2015-parameters/src/rest.js [hunk 1]
```diff
@@ -107,9 +107,17 @@ function hasRest(node) {
 }
 
 function optimiseIndexGetter(path, argsId, offset) {
+  let index;
+
+  if (t.isNumericLiteral(path.parent.property)) {
+    index = t.numericLiteral(path.parent.property.value + offset);
+  } else {
+    index = t.binaryExpression("+", path.parent.property, t.numericLiteral(offset));
```
Comment (hzoo) [inline packages/babel-plugin-transform-es2015-parameters/src/rest.js]: Can we do like `index = t.numericLiteral(path.parent.property + offset);`?

Comment (hzoo) [inline packages/babel-plugin-transform-es2015-parameters/src/rest.js]: Although probably best to leave it since it's in the else. :+1:

Comment (amasad) [inline packages/babel-plugin-transform-es2015-parameters/src/rest.js]: It's resulting in `NaN` because it's object + number

**Comment by kpdecker** [comment]
@amasad thank you for the quick fix on this.

File: packages/babel-plugin-transform-es2015-parameters/test/fixtures/parameters/rest-length/expected.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+var t = function (f) {
+  arguments.length <= 1 ? undefined : arguments[1];
+  arguments.length <= arguments.length - 1 - 1 + 1 ? undefined : arguments[arguments.length - 1 - 1 + 1];
```
Comment (kpdecker) [inline packages/babel-plugin-transform-es2015-parameters/test/fixtures/parameters/rest-length/expected.js]: Could his be optimized out or is the assumption that uglify will reduce it back to the simplified `args[args.length - 1]` form?

Comment (amasad) [inline packages/babel-plugin-transform-es2015-parameters/test/fixtures/parameters/rest-length/expected.js]: Yeah the minifier should help, but we can also do some evaluation here. If
you want to try it, check out path.evaluate

Comment (kpdecker) [inline packages/babel-plugin-transform-es2015-parameters/test/fixtures/parameters/rest-length/expected.js]: Sure, sounds like a fun challenge :)

**Comment by fabiomcosta** [comment]
cool, thx for working on this @amasad and @kpdecker !

**Pull request merged**